$OpenBSD: patch-tools_lld_ELF_Driver_cpp,v 1.2 2017/03/18 20:49:42 ajacoutot Exp $

- ELF: Add /usr/lib as default search path.

  GNU ld and gold have this path as default search path.
  If you don't want this directory, pass -nostdlib.
- Enable PIE by default.
- Add support for -znodlopen.

--- tools/lld/ELF/Driver.cpp.orig	Wed Feb  1 17:45:57 2017
+++ tools/lld/ELF/Driver.cpp	Sat Mar 18 15:46:22 2017
@@ -459,6 +459,9 @@ static std::vector<StringRef> getLines(MemoryBufferRef
 
 // Initializes Config members by the command line options.
 void LinkerDriver::readConfigs(opt::InputArgList &Args) {
+  if (!Args.hasArg(OPT_nostdlib))
+    Config->SearchPaths.push_back("=/usr/lib");
+
   for (auto *Arg : Args.filtered(OPT_L))
     Config->SearchPaths.push_back(Arg->getValue());
 
@@ -493,7 +496,8 @@ void LinkerDriver::readConfigs(opt::InputArgList &Args
   Config->NoUndefinedVersion = Args.hasArg(OPT_no_undefined_version);
   Config->Nostdlib = Args.hasArg(OPT_nostdlib);
   Config->OMagic = Args.hasArg(OPT_omagic);
-  Config->Pie = getArg(Args, OPT_pie, OPT_nopie, false);
+  Config->Pie = getArg(Args, OPT_pie, OPT_nopie,
+      !Args.hasArg(OPT_shared) && !Args.hasArg(OPT_relocatable));
   Config->PrintGcSections = Args.hasArg(OPT_print_gc_sections);
   Config->Relocatable = Args.hasArg(OPT_relocatable);
   Config->DefineCommon = getArg(Args, OPT_define_common, OPT_no_define_common,
@@ -532,6 +536,7 @@ void LinkerDriver::readConfigs(opt::InputArgList &Args
   Config->ZCombreloc = !hasZOption(Args, "nocombreloc");
   Config->ZExecstack = hasZOption(Args, "execstack");
   Config->ZNodelete = hasZOption(Args, "nodelete");
+  Config->ZNodlopen = hasZOption(Args, "nodlopen");
   Config->ZNow = hasZOption(Args, "now");
   Config->ZOrigin = hasZOption(Args, "origin");
   Config->ZRelro = !hasZOption(Args, "norelro");
